unity/Hello_World/Assets/Scripts/SpatialMappingCapture.cs

```diff
@@ -0,0 +1,250 @@
+//public class SpatialMappingCapture : MonoBehaviour
+public class SpatialMappingCapture : MonoBehaviour, IMixedRealitySpatialAwarenessObservationHandler<SpatialAwarenessMeshObject>
+{
+    System.Net.Sockets.TcpClient tcpClient;
+    System.Net.Sockets.TcpListener tcpServer;
+    NetworkStream tcpStream;
+    public string TcpServerIPAddr = "";
+    public const int SMTcpPort = 11010;
+
+    private Logger _logger = null;
+
+    string debugString = "";
+
+    // Spatial awareness & scene understanding
+    //List<MixedRealitySpatialAwarenessEventData<SpatialAwarenessMeshObject>> meshes = new List<MixedRealitySpatialAwarenessEventData<SpatialAwarenessMeshObject>>();
+    //IMixedRealitySpatialAwarenessMeshObserver spatialAwarenessObserver = null;
+    //Scene worldScene;
+    //private IMixedRealitySceneUnderstandingObserver sceneUnderstandingObserver;
+
+    /// <summary>
+    /// Lazy acquire the logger object and return the reference to it.
+    /// </summary>
+    /// <returns>Logger instance reference.</returns>
+    private ref Logger logger()
+    {
+        if (this._logger == null)
+        {
+            // TODO: Error handling for null loggerObject?
+            this._logger = GameObject.Find("Logger").GetComponent<Logger>();
+        }
+        return ref this._logger;
+    }
+
+    // Start is called before the first frame update
+    protected void Start()
+    {
+        Logger log = logger();
+
+        // Suspend Mesh Observation from all Observers until we connect to the TCP socket
+        CoreServices.SpatialAwarenessSystem.SuspendObservers();
+
+        try
+        {
+            TcpServerIPAddr = PTGUtilities.getIPv4AddressString();
+        }
+        catch (InvalidIPConfiguration e)
+        {
+            log.LogInfo(e.ToString());
+            return;
+        }
+
+        Thread tSpatialMappingCapture = new Thread(SetupSpatialMappingCapture);
+        tSpatialMappingCapture.Start();
+        //log.LogInfo("Waiting for spatial mapping TCP connection");
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (debugString != "")
+        {
+            this.logger().LogInfo(debugString);
+            debugString = "";
+        }
+
+        if (tcpStream != null)
+        {
+            // Resume Mesh Observation from all Observers
+            CoreServices.SpatialAwarenessSystem.ResumeObservers();
+        }
+    }
+
+    void SetupSpatialMappingCapture()
+    {
+        IPAddress localAddr = IPAddress.Parse(TcpServerIPAddr);
+
+        tcpServer = new TcpListener(localAddr, SMTcpPort);
+
+        // Start listening for client requests.
+        tcpServer.Start();
+
+        // Perform a blocking call to accept requests.
+        // You could also use server.AcceptSocket() here.
+        tcpClient = tcpServer.AcceptTcpClient();
+        tcpStream = tcpClient.GetStream();
```

Comment:
Not really a comment about this PR, however could this function, which seems to be forked off to a thread to process, also continue on heart-beat or something to check for new client connections? I'm thinking of being robust to the ROS node receiver crashing / shutting-down / restarting use-case.

Reply:
I think that could work. I made it its own thread because if it's on the main unity thread, the `AcceptTCPClient()` call will block the thread until the TCP connection is established, which means nothing is rendered to the headset and it looks like nothing is happening.

Reply:
> I made it its own thread ...

No argument there, more thinking that this separate thread could continue existing and loop, re-attempting tcp client acceptance after the current one "closes", assuming that is appropriately detected.

Of course, it's likely the case that the potential effort doing that should be expended moving to something like 0mq/redis/etc. that will handle the communications (including reconnections, etc.) for us.